daemon/containerd/image.go

```diff
@@ -44,19 +44,33 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 	}
 
 	cs := i.client.ContentStore()
-	conf, err := containerdimages.Config(ctx, cs, desc, platform)
-	if err != nil {
-		return nil, err
-	}
 
-	imageConfigBytes, err := content.ReadBlob(ctx, cs, conf)
+	var manifests []ocispec.Manifest
+	err = i.walkImageManifests(ctx, desc, func(img *ImageManifest) error {
+		mfst, err := img.Manifest(ctx)
```

Comment:
If we don't need the full `ocispec.Manifest`, we can just directly read the config with `img.Config(ctx)`.